artemis-core-client/src/main/java/org/apache/activemq/artemis/utils/ObjectInputStreamWithClassLoader.java

```diff
@@ -25,23 +25,110 @@
+
    // Attributes -----------------------------------------------------------------------------------
 
+   private List<String> whiteList = new ArrayList<>();
+   private List<String> blackList = new ArrayList<>();
+
    // Static ---------------------------------------------------------------------------------------
 
    // Constructors ---------------------------------------------------------------------------------
 
    public ObjectInputStreamWithClassLoader(final InputStream in) throws IOException {
       super(in);
+      String whiteList = System.getProperty(WHITELIST_PROPERTY, null);
+      setWhiteList(whiteList);
+
+      String blackList = System.getProperty(BLACKLIST_PROPERTY, null);
+      setBlackList(blackList);
    }
 
    // Public ---------------------------------------------------------------------------------------
 
+   /**
+    * @return the whiteList configured on this policy instance.
+    */
+   public String getWhiteList() {
+      Iterator<String> entries = whiteList.iterator();
+      StringBuilder builder = new StringBuilder();
+
+      while (entries.hasNext()) {
+         builder.append(entries.next());
+         if (entries.hasNext()) {
+            builder.append(",");
+         }
+      }
+      return builder.toString();
+   }
+
+   /**
+    * @return the blackList configured on this policy instance.
+    */
+   public String getBlackList() {
+      Iterator<String> entries = blackList.iterator();
+      StringBuilder builder = new StringBuilder();
+
+      while (entries.hasNext()) {
+         builder.append(entries.next());
+         if (entries.hasNext()) {
+            builder.append(",");
+         }
+      }
+
+      return builder.toString();
+   }
+
+   /**
+    * Replaces the currently configured whiteList with a comma separated
+    * string containing the new whiteList. Null or empty string denotes
+    * no whiteList entries, {@value #CATCH_ALL_WILDCARD} indicates that
+    * all classes are whiteListed.
+    *
+    * @param whiteList the whiteList that this policy is configured to recognize.
+    */
+   public void setWhiteList(String whiteList) {
+      ArrayList<String> list = new ArrayList<>();
+      if (whiteList != null && !whiteList.isEmpty()) {
+         list.addAll(Arrays.asList(whiteList.split(",")));
+      }
+
+      this.whiteList = list;
+   }
+
+   /**
+    * Replaces the currently configured blackList with a comma separated
+    * string containing the new blackList. Null or empty string denotes
+    * no blacklist entries, {@value #CATCH_ALL_WILDCARD} indicates that
+    * all classes are blacklisted.
+    *
+    * @param blackList the blackList that this policy is configured to recognize.
+    */
+   public void setBlackList(String blackList) {
+      ArrayList<String> list = new ArrayList<>();
+      if (blackList != null && !blackList.isEmpty()) {
+         list.addAll(Arrays.asList(blackList.split(",")));
+      }
+
+      this.blackList = list;
+   }
+
```

Comment:
Same comment as above.  You could move the list validation and parse into a separation method to avoid duplication.